File: backend/app/schemas/opportunity.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class OpportunityCreateRequest(BaseModel):
    title: str
    description: str
    opportunity_type: str
    location: str
    skills_required: str
    application_deadline: datetime


class OpportunityUpdateRequest(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    opportunity_type: Optional[str] = None
    location: Optional[str] = None
    skills_required: Optional[str] = None
    application_deadline: Optional[datetime] = None
    is_active: Optional[bool] = None


class OrganizationSummary(BaseModel):
    user_id: int
    organization_name: str
    industry: str
    website: str
    description: str
    location: str
    verified: bool

    class Config:
        from_attributes = True


class OpportunityResponse(BaseModel):
    id: int

    title: str
    description: str

    opportunity_type: str
    location: str

    skills_required: str

    application_deadline: datetime

    is_active: bool
    created_at: datetime

    organization: OrganizationSummary

    class Config:
        from_attributes = True

File: frontend/src/types/opportunity.ts
export interface OrganizationSummary {
  user_id: number;

  organization_name: string;
  industry: string;

  website: string;
  description: string;

  location: string;
  verified: boolean;
}

export interface Opportunity {
  id: number;

  title: string;
  description: string;

  opportunity_type: string;
  location: string;

  skills_required: string;

  application_deadline: string;

  is_active: boolean;
  created_at: string;

  organization: OrganizationSummary;
}

export interface CreateOpportunityRequest {
  title: string;
  description: string;
  opportunity_type: string;
  location: string;
  skills_required: string;
  application_deadline: string;
}

export interface UpdateOpportunityRequest {
  title?: string;
  description?: string;
  opportunity_type?: string;
  location?: string;
  skills_required?: string;
  application_deadline?: string;
  is_active?: boolean;
}